packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -109,4 +138,37 @@ export class WorkspaceFrontendContribution implements CommandContribution, MenuC
         }
     }
 
+    protected async saveWorkspaceAs(): Promise<void> {
+        let overwrite: boolean;
+        let selected: URI | undefined;
+        do {
+            selected = await this.fileDialogService.showSaveDialog({
+                title: WorkspaceCommands.SAVE_WORKSPACE_AS.label!,
+                filters: {
+                    'Theia Workspace (*.theia-workspace)': [THEIA_EXT],
+                    'VS Code Workspace (*.code-workspace)': [VSCODE_EXT]
+                }
+            });
+            overwrite = await this.confirmOverwrite(selected);
+        } while (!overwrite);
+
+        if (selected) {
+            const displayName = selected.displayName;
+            if (!displayName.endsWith(`.${THEIA_EXT}`) && !displayName.endsWith(`.${VSCODE_EXT}`)) {
+                selected = selected.parent.resolve(`${displayName}.${THEIA_EXT}`);
```

Comment:
it looks strange, you ask a confirmation for one file name, but then save with another

Reply:
in this way, the users don't have to key-in the file extension every time - `theia-workspace` isn't a short string :) vsCode does the same thing.

Reply:
it's fine, I meant your check that a file `blabla` does not exist but then save with `blabla.theia-workspace`.

Reply:
you are right. I changed the logic to make sure the `filesystem.exist` and `workspaceService.save` are targeting the same file.